analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs

```diff
@@ -137,12 +137,17 @@ public static bool IsSecurityHotspot(DiagnosticDescriptor diagnostic)
                 ? NuGetMetadataReference.MSTestTestFrameworkV1  // Any reference to detect a test project
                 : Enumerable.Empty<MetadataReference>();
 
-        public static AnalyzerOptions CreateOptions(string relativePath)
+        public static AnalyzerOptions CreateOptions(string relativePath, SourceText text = null)
```

Comment:
A bit hacky. Used in the ParameterLoaderTest `SetParameterValues_CalledTwiceAfterChangeInConfigFile_UpdatesProperties`

Reply:
It might be neater to make the `text` parameter a string and to split the method into two e.g.

`CreateOptions(string relativePath)` and 
`CreateOptions(string path, string content)`.

The second method would just create and return the options based on the supplied data.

The first would contain the old "_use the content from the file if it exists_" logic.